src/main/java/com/rtbhouse/utils/avro/FastDeserializerGenerator.java

```diff
@@ -267,11 +290,151 @@ private void processRecord(JVar schemaVariable, JVar containerVariable, final Sc
+                    JVar schemaVar = null;
+                    if (useGenericTypes) {
+                        schemaVar = declareSchemaVariableForRecordField(readerField.name(), readerField.schema(),
+                                schemaVariable);
+                    }
+                    JExpression value = parseDefaultValue(readerField.schema(), readerField.defaultValue(), body,
+                            schemaVar, readerField.name());
+                    body.invoke(result, "put").arg(JExpr.lit(readerField.pos())).arg(value);
+                }
+            }
+        }
+
         if (recordAction.getShouldRead()) {
             body._return(result);
         }
     }
 
+    private JExpression parseDefaultValue(Schema schema, JsonNode defaultValue, JBlock body, JVar schemaVariable,
+            String fieldName) {
+        Schema.Type schemaType = schema.getType();
+
+        // Default value of union have to have first type from union
+        if (schemaType == Schema.Type.UNION) {
+            schema = schema.getTypes().get(0);
+            schemaType = schema.getType();
+            schemaVariable = declareSchemaVariableForUnion(fieldName, schema, schemaVariable, 0);
+        }
+
+        if (schemaType == Schema.Type.RECORD) {
+            JType recordType;
+            JVar recordVar;
+            if (!useGenericTypes) {
+                recordType = codeModel.ref(schema.getFullName());
+                recordVar = body.decl(recordType, getVariableName("default" + schema.getName()),
+                        JExpr._new(recordType));
+            } else {
+                recordType = codeModel.ref(GenericData.Record.class);
+                recordVar = body.decl(recordType, getVariableName("default" + schema.getName()),
+                        JExpr._new(recordType).arg(schemaMapField.invoke("get").arg(JExpr.lit(getSchemaId(schema)))));
+            }
+            for (Iterator<Map.Entry<String, JsonNode>> it = defaultValue.getFields(); it.hasNext(); ) {
+                Map.Entry<String, JsonNode> subFieldEntry = it.next();
+                String subFieldName = subFieldEntry.getKey();
+                Schema.Field subField = schema.getField(subFieldName);
+                JsonNode value = subFieldEntry.getValue();
+
+                int fieldNumber = subField.pos();
+
+                JVar schemaVar = declareSchemaVariableForRecordField(subField.name(), subField.schema(),
+                        schemaVariable);
+                JExpression fieldValue = parseDefaultValue(subField.schema(), value, body, schemaVar, subField.name());
+                body.invoke(recordVar, "put").arg(JExpr.lit(fieldNumber)).arg(fieldValue);
+            }
+            return recordVar;
+
+        } else if (schemaType == Schema.Type.ARRAY) {
+            Schema elementSchema = schema.getElementType();
+            JVar arrayVar = body.decl(codeModel.ref(ArrayList.class), getVariableName("defaultArray"),
```

Comment:
It's better not to use raw types. It's preferred to use parameterized types - hint: "narrow" method from code model API. Apart from that, Apache Avro uses a specific array type - GenericData.Array, which additionally requires element schema parameter. It is better to use this type, at least for generic data records (where element schema is easily available).

Reply:
Right, it was the cause of unchecked warnings - now it's fixed.